contrib/IECoreUSD/src/IECoreUSD/ShaderAlgo.cpp

```diff
@@ -278,40 +333,18 @@ pxr::UsdShadeOutput IECoreUSD::ShaderAlgo::writeShaderNetwork( const IECoreScene
 			// Currently, we don't really track output types in Gaffer.
 			networkOutUsd = usdShader.CreateOutput( outName, pxr::SdfValueTypeNames->Token );
 		}
-
-		const IECore::BoolData* adapterMeta = shader.second->blindData()->member<IECore::BoolData>( IECoreScene::ShaderNetworkAlgo::componentConnectionAdapterLabel() );
-		if( adapterMeta && adapterMeta->readable() )
-		{
-			usdShader.GetPrim().SetMetadata( g_adapterLabelToken, true );
-		}
 	}
 
-	for( const auto &shader : shaderNetworkWithAdapters->shaders() )
-	{
-		pxr::UsdShadeShader usdShader = pxr::UsdShadeShader::Get( shaderContainer.GetStage(), shaderContainer.GetPath().AppendChild( pxr::TfToken( pxr::TfMakeValidIdentifier( shader.first.string() ) ) ) );
-		for( const auto &c : shaderNetworkWithAdapters->inputConnections( shader.first ) )
-		{
-			pxr::UsdShadeInput dest = usdShader.GetInput( pxr::TfToken( c.destination.name.string() ) );
-			if( ! dest.GetPrim().IsValid() )
-			{
-				dest = usdShader.CreateInput( toUSDParameterName( c.destination.name ), pxr::SdfValueTypeNames->Token );
-			}
-
-			pxr::UsdShadeShader sourceUsdShader = pxr::UsdShadeShader::Get( shaderContainer.GetStage(), shaderContainer.GetPath().AppendChild( pxr::TfToken( pxr::TfMakeValidIdentifier( c.source.shader.string() ) ) ) );
-			std::string sourceOutputName = c.source.name.string();
-			if( sourceOutputName.size() == 0 )
-			{
-				sourceOutputName = "DEFAULT_OUTPUT";
-			}
-			pxr::UsdShadeOutput source = sourceUsdShader.CreateOutput( pxr::TfToken( sourceOutputName ), dest.GetTypeName() );
-			dest.ConnectToSource( source );
-		}
-
-	}
+	writeShaderConnections( shaderNetwork, usdShaders );
 
 	return networkOutUsd;
 }
 
+// This is very similar to `writeShaderNetwork` but with these key differences :
+//
+// - The output shader is written as a UsdLight-derived prim rather than a UsdShadeShader.
+// - The other shaders are parented inside the light.
+// - We don't need to create a UsdShadeOutput to return.
```

Comment:
This comment seems to belong with  writeLight, not readShaderNetwork?

Reply:
Thanks for spotting that - fixed in https://github.com/ImageEngine/cortex/pull/1376/commits/f174f8d62386e4c1b84324b698796617e4aa8ac0.